src/orangepi/ProductionFiles/FinalPiProgram.py

```diff
@@ -179,14 +145,169 @@ def average_position_of_pixels(mat, threshold=128):
+            parseError("camera '{}': could not read path".format(cam.name))
+            return False
+
+        # stream properties
+        cam.streamConfig = config.get("stream")
+
+        cam.config = config
+
+        cameraConfigs.append(cam)
+        return True
+
+    def readSwitchedCameraConfig(config):
+        """Read single switched camera configuration."""
+        cam = CameraConfig()
+
+        # name
+        try:
+            cam.name = config["name"]
+        except KeyError:
+            parseError("could not read switched camera name")
+            return False
+
+        # path
+        try:
+            cam.key = config["key"]
+        except KeyError:
+            parseError("switched camera '{}': could not read key".format(cam.name))
+            return False
+
+        switchedCameraConfigs.append(cam)
+        return True
+
+    def readConfig():
+        """Read configuration file."""
+        global team
+        global server
+
+        # parse file
+        try:
+            with open(configFile, "rt", encoding="utf-8") as f:
+                j = json.load(f)
+        except OSError as err:
+            parseError("could not open '{}': {}".format(configFile, err), file=sys.stderr)
+            return False
+
+        # top level must be an object
+        if not isinstance(j, dict):
+            parseError("must be JSON object")
+            return False
+
+        # team number
+        try:
+            team = 5607
+        except KeyError:
+            parseError("could not read team number")
+            return False
+
+        # ntmode (optional)
+        if "ntmode" in j:
+            str = j["ntmode"]
+            if str.lower() == "client":
+                server = False
+            elif str.lower() == "server":
+                server = True
+            else:
+                parseError("could not understand ntmode value '{}'".format(str))
+
+        # cameras
+        try:
+            cameras = j["cameras"]
+        except KeyError:
+            parseError("could not read cameras")
+            return False
+        for camera in cameras:
+            if not readCameraConfig(camera):
+                return False
+
+        # switched cameras
+        if "switched cameras" in j:
+            for camera in j["switched cameras"]:
+                if not readSwitchedCameraConfig(camera):
+                    return False
+
+        return True
+
+    def startCamera(config):
+        """Start running the camera."""
+        print("Starting camera '{}' on {}".format(config.name, config.path))
+        camera = UsbCamera(config.name, config.path)
+        server = CameraServer.startAutomaticCapture(camera=camera)
+
+        camera.setConfigJson(json.dumps(config.config))
+        camera.setConnectionStrategy(VideoSource.ConnectionStrategy.kConnectionKeepOpen)
+
+        if config.streamConfig is not None:
+            server.setConfigJson(json.dumps(config.streamConfig))
+
+        return camera
+
```

Comment:
I recommend the CS.enableLogging() from the myCamPub.py to enable the logging within the CameraServer code.